2016/05/what-is-react.js-and-why-use-react.md

```diff
@@ -0,0 +1,41 @@
+I have been using [React.js](https://facebook.github.io/react/index.html), a JavaScrip library for creating user interfaces, for about 4 months now.  You might want to read the official [getting-started](https://facebook.github.io/react/docs/getting-started.html) or [Tutorial](https://facebook.github.io/react/docs/tutorial.html) pages.
+
+**React has the following advantages:**
+
+* Good if you what to build large applications with data that changes over time.
+* **Virtual DOM**. React abstracts away the DOM, giving a simple programming model and better performance.
+* Implements one-way reactive data flow.
+* Is based in components.
+* Is the V in MVC.
+* You can always change how the component will render looking at one source file.
+* Building Javascript and HTML into JSX makes components easily understandable.
+* React.js is declarative, which means that React conceptually hits the “refresh” button any time data changes.
+You can use React.js seamlessly with Node.js or AngularJS.
+
+This is an component example:
+
+`var HelloMessage = React.createClass({
+    render: function() {
+      return <div>Hello {this.props.name}</div>;
+    }
+  });
+  ReactDOM.render(<HelloMessage name="John" />, mountNode);`
+
+
+**Recommendations:**
+
+* Use [React.js](https://facebook.github.io/react/index.html) only if your application have very much dynamic page updating.
+* Break the UI into a component hierarchy.
+Use [Redux](http://redux.js.org/) instead of [Flux](https://facebook.github.io/flux/).
+* Write test of each of your components
+* Use JSX, ES6, Babel, Webpack, and NPM.
```

Comment:
Add why you should use each of the tools you mention here and previously on this post.